[8.0] mrp_hook + mrp_operations_extension

pedrobaeza opened the pull request:
This PR includes 2 modules because the first is only a technical one for providing hooks for allowing to work properly the other one (and others that will come later).

This is the README for the main one:
# 
# MRP operations extension module

This module adds:
- New table to store operations to avoid typing them again.
- Adds a relation from workcenter lines to BoM lists.
- Adds a relation from workcenter lines to Manufacturing Orders in
  Scheduled/Consumed/Finished products.
- Adds a relation between routing workcenter lines and workcenter extra info.
- Allows to set specific times per routing in addition to workcenter times.
- Controls the availability of material in operations
- Controls operation beginning in previous operations are not finished.
# Configuration

Go to _Settings > Manufacturing_ and activate "Manage routings and work orders"
for handling the features of this module.

Commit:
[ADD] mrp_operations_extension

Commit:
[MOD] view changes

pedrobaeza committed:
[IMP] mrp_operations_extension: PEP8 + Travis

oihane committed:
[FIX+IMP] mrp_operations_extension

Commit:
[FIX] traceback al confirmar una MO

oihane committed:
[FIX] Pylint fixing <mrp_operations_extension>

Commit:
[FIX] mrp_operations_extension: When no routing_id is defined in the manufacturing order move Final Product to Stock check in operation is not controlled.

oihane committed:
[FIX] mrp_operations_extension

Phantom bom lines with workorder association
When installing variants, it doesn't assign corresponding wo to scheduled product
Domain added to avoid selecting workorders that don't apply

Commit:
[IMP] mrp_operations_extension: Limit to not add products from work order tree

oihane committed:
[FIX+IMP] mrp_operations_extension

IMP: Deleting workers info and new module <mrp_workcenter_workers_operations>
IMP: Adding fields in base modules, extension updates onchange
IMP: Changes in views and field names
REM: Deleted <mrp_workcenter_workers> module
IMP: Configuration part added

pedrobaeza committed:
[IMP] mrp_operations_extension: Cost labels

oihane committed:
[FIX+IMP] mrp_operations_extension

IMP: Translations
FIX: Fixing some names
FIX: Some functional changes
FIX: Default line functionality

pedrobaeza committed:
[FIX+IMP] mrp_operations_extension

FIX: Operation where is production properly handled
IMP: Some views enhanced
FIX: No warning at MO confirmation if there is no route.
IMP: Show cycle length and capacity in route.
FIX: Propagate capacity data.

Commit:
[FIX+IMP] mrp_operations_extension

FIX: correct actions view reference on produce button
IMP: Changes in the calculation of number of cycles + hours
IMP: Group analytic lines
IMP: Remove product check
FIX: Wizard when produced from work order
IMP: Change product cost computation

pedrobaeza committed:
[FIX+IMP] mrp_operations_extension:

FIX: Use ceil instead is_integer check
IMP: Don't allow inline editing of route operations on BoM

Commit:
[FIX] Check for track_all

Commit:
[IMP] File name consistency with model name

Commit:
[FIX] call super before testing the workcenters

We check that the production order's workcenter lines contain at least
one "Produce here" line", but at this stage they are not created because
we didn't call super().
So let's call super() first.

oihane committed:
[FIX] Translation fixed because of issue #660

pedrobaeza committed:
[FIX+IMP] mrp_operations_extension

IMP: Show workcenter on route operation if there are no lines in workcenters.
FIX: Fix workcenter_id invisible condition in routing

Commit:
[IMP] mrp_operations_extension: Has gotten mrp_operation_produce programming module. closses #644.

Commit:
[IMP] <mrp_operations_extension>: Modifications added to avoid errors in miss configurations.

oihane committed:
[MOD] Only first line default, if written

Commit:
[FIX] mrp_operations_extension: correct translations, and improved programming.

pedrobaeza committed:
[IMP] Expand AUTHORS

Commit:
[FIX] mrp_operations_extension: permite iniciar una OF con líneas en estado cancelado, realizado o asignado

sysadminmatmoz committed:
Slovene translations added

pedrobaeza committed:
[FIX] mrp_operations_extension: Added hr dependency. Closes #753. Retrieve correctly work center

Commit:
[IMP] mrp_operations_extension: Added material ready for operations + editable operations from production order

Commit:
[ADD] mrp_configurable_timing: modulo que hace configurable los decimales de tiempos de ciclo
[ADD] mrp_product_variants_configurable_timing: enlaza mrp_configurable_timing + mrp_operations_extension
[IMP] mrp_operations_extension + mrp_product_variants_operations: calculo de ciclos modificados.

Commit:
[FIX+IMP] mrp_operations_extension

FIX: Fixes #920 Now initiate a WO doesn't force MO material reservation
IMP: Creation of the field "picking_type_id" in mrp.routing.workcenter.
FIX: Closes #921
IMP: Added function to avoid repeat code
FIX: When WO starts the MO state should be 'in_production' not 'ready'

oihane committed:
[IMP] <mrp_operations_extension> Added some demo data

oihane committed:
[FIX] <mrp_operations_extension> Fixed workcenter open issue closes #951

pedrobaeza committed:
[IMP] mrp_operations_extension: Label for "Operator average hour cost"

alfredoavanzosc committed:
[FIX] mrp_operations_extension: Fixed "_operators_number_avg_cost" function, of the model "mrp.production.workcenter.line", for calculating fields: '# Operators' and 'Operator average hour cost'.
[IMP] mrp_operations_extension: Put a "onchange" to calculate fields # Operators, and Operator average hour cost, of model mrp.workcenter.

Commit:
[IMP] mrp_operations_extension: nuevas funcionalidades en workorders.
Closes #750 #812 - Se han realizado solo algunos puntos del issue ya que algunos de ellos se han solucionado en las nuevas actualizaciones
	- nueva vista calendario, según fechas previstas
	- permitir modificar la fecha planificada de una OF si no está iniciada
	- cambiar la fecha prevista de una OF cuando se cambia la fecha prevista de una OT si esta es menor que la de su OF

Commit:
[FIX] mrp_operations_extension: Fixed bug infineloop when updating date_planned

pedrobaeza committed:
OCA Transbot updated translations from Transifex

StefanRijnhart committed:
[ADD] Set default for do_production upon installation

pedrobaeza committed:
[IMP] mrp_hook: Provide hooks for _bom_explode method

As we are struggled by the structure of core current code, and
https://github.com/odoo/odoo/pull/8885 hasn't been accepted in v8,
we need to make this hard move to allow a lot of workarounds like
modules not depending this one, but anyway the method itself is
not too much inheritable.

pedrobaeza committed:
[IMP] mrp_hook: Provide hooks for _bom_explode method

As we are struggled by the structure of core current code, and
https://github.com/odoo/odoo/pull/8885 hasn't been accepted in v8,
we need to make this hard move to allow a lot of workarounds like
modules not depending this one, but anyway the method itself is
not too much inheritable.

pedrobaeza committed:
[IMP] mrp_operations_extension: Fine tuning of UI + computations

* Guided selection of operation inside routing lines for avoiding confusions.
* Homogenization of work centers selection in routing lines. You always add
  a workcenter line.
* Limit selectable workcenters on the work order according routing line.
* Views cleaning.

pedrobaeza committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] mrp_hook+mrp_operations_extension: Adaptation to OCA

oihane commented:
@pedrobaeza demo data is causing travis fail:

```
ParseError: "hr.employee" while parsing /home/travis/build/OCA/manufacture/mrp_operations_extension/demo/hr_employee_demo.xml:5, near
<record id="hr.employee_al" model="hr.employee">
    <field name="user_id" ref="mrp_operations_extension.user_opeext_1"/>
</record>
```

oihane commented:
**mrp_operation_extension** requires dependency with **hr**

rafaelbn commented:
Could anyone add label need review please?

nhomar commented:
Hello thanks.

Comment 1.- Please check which code is not tested in order to a little tests to achieve  a better coverage.

nhomar commented on mrp_hook/models/mrp_bom.py:
```diff
@@ -0,0 +1,150 @@
+    _inherit = 'mrp.bom'
+
+    @api.model
+    def _factor(self, factor, product_efficiency, product_rounding):
```
Can you document this a little better, I understand that you can force a factor, but "why"?

factor / NUMBER NEAR TO 0 == INFINITE , why the denominator is 1.0 should not be forced the factor = 0?

pedrobaeza replied:
This is a verbatim copy of the core one

nhomar replied:
Cool, can you put that on the docstring please?:

```
verbatin copy from here _"link to code"
```

pedrobaeza replied:
Well, the entire module is a verbatim copy with the corresponding hooks. That's why there isn't tests (not in this module), and the documentation is in the README, but I can be more explicit..

nhomar commented on mrp_operations_extension/__init__.py:
```diff
@@ -0,0 +1,33 @@
+                        'custom_data': False})]
+
+
+def post_init_hook(cr, pool):
```
Can you documment this method to toe README in order to inform users which are checking the documentation that the data structure and the real data will change once this module is installed?.

nhomar commented:
It seems a cool set of modules conceptually +1.

I should recommend several things in order to ensure better review process:

1.- A little video with the before - after this 2 modules.
2.- More tests, I think there are too little test cases (but this is an impression), I think this 2 modules are like the base of everything then we need to chck the traceback even the init hook should be tested, becaus in an error this module will change a lot data in an actual MRP environment.
3.- The init hook should have a backup strategy in order to ensure data changed is backuped (this should be an extra not mandatory for my +1)

Thanks a lot for this kind of work.

moylop260 commented:
Could you check next [result build](https://travis-ci.org/OCA/manufacture/jobs/93542578#L266-L267)?

``` bash
************* Module mrp_hook.__openerp__
mrp_hook/__openerp__.py:5: [C8101(manifest-required-author), ] Missing author required "Odoo Community Association (OCA)" in manifest file
mrp_hook/__openerp__.py:5: [C8102(manifest-required-key), ] Missing required key "license" in manifest file
mrp_hook/__openerp__.py:5: [C8102(manifest-required-key), ] Missing required key "images" in manifest file
************* Module mrp_operations_extension
mrp_operations_extension/__init__.py:1: [W7902(duplicate-xml-record-id), ] Duplicate xml record id mrp_operations_extension.req_link_mrp_workcenter.noupdate=0
************* Module mrp_operations_extension.__openerp__
mrp_operations_extension/__openerp__.py:1: [C8201(no-utf8-coding-comment), ] No UTF-8 coding comment found: Use `# coding: utf-8` or `# -*- coding: utf-8 -*-`
mrp_operations_extension/__openerp__.py:19: [C8102(manifest-required-key), ] Missing required key "license" in manifest file
mrp_operations_extension/__openerp__.py:19: [C8102(manifest-required-key), ] Missing required key "images" in manifest file
```

JMODaERP commented:
Hello Pedro:

Step 1: Add a relation from workcenter lines to BoM lists.

**I have done step 1 and have enabled the BOM**.

Step 2:  Add a relation from workcenter lines to Manufacturing Orders in Scheduled/Consumed/Finished products.

**What do you mean: Adds a relation from workcenter lines to Manufacturing Orders in Scheduled / Consumed / Finished products ?**. When I click "Compute Data", the system automatically enters the workplace and the need to produce material.

elicoidal commented on mrp_hook/__openerp__.py:
```diff
@@ -0,0 +1,22 @@
+    "contributors": [
+        "Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>",
+    ],
+    "depends": [
```
Please remove empty keys

elicoidal commented on mrp_hook/models/mrp_bom.py:
```diff
@@ -0,0 +1,150 @@
+# (c) 2015 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, tools, _
```
Please sort in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

elicoidal commented on mrp_operations_extension/README.rst:
```diff
@@ -0,0 +1,76 @@
+This module adds:
+
+* New table to store operations to avoid typing them again.
+* Adds a relation from workcenter lines to BoM lists.
```
"A relation between workcenter lines and BoM lists."

elicoidal commented on mrp_operations_extension/README.rst:
```diff
@@ -0,0 +1,76 @@
+
+* New table to store operations to avoid typing them again.
+* Adds a relation from workcenter lines to BoM lists.
+* Adds a relation from workcenter lines to Manufacturing Orders in
```
"A relation between workcenter lines and Manufacturing Orders in
  Scheduled/Consumed/Finished products."

JMODaERP commented:
Well, talking about relations in Odoo, for each product there are one and only one bill of materials.

![2015-11-29 19_13_36-sin titulo - paint](https://cloud.githubusercontent.com/assets/15608043/11459079/72676c40-96ce-11e5-81b9-22437860722e.jpg)

A Bill of Row Materials (BoM) can contain one o more Row Material, but only one Routing (production line), we could say that BoM is one table. Whose line of production is only and its primary key is the product. The BoM contains one foreing key for each row material an only one for routing (contains one foreing key for each row material and only one for routing (routing, productive line and productive process are the same concept).

In Odoo, Work Order is a relation between Operation and end product, both takes its name. Wok order is performed in a Work Center, and the capacity of the Work Center is predefined in itself, unless, otherwise directed by a work order.

Therefore a table is necessary or at least a view to store and then display, the actual capacity of the Centre for a work order and the hours provided for her in particular.

For example, if the machinery of a work center consists of a saw, does not require the same effort cutting timber that polystyrene foams, and therefore has less real capacity for the first thing for the second and that means more running time for the same operation, cut.

Do now, again I ask, what I have to prove?. What test requires the PR

anajuaristi commented:
Hi Jose Manuel:

The following sentence is not true... for each product there are one and
only one bill of materials. Relation is one product, several bill of
materials, where each bill of materials contains a routing.
Routing contains routing steps. In standar Odoo one routing step contains a
single machine but in OdooMRP we added the option to have several machines
on each routing step containing each one their own capacity related to a
routing line.

Odoo standar structure has been strongly extended by OdooMRP so IMHO your
relational graphic is not exactly the one we have.

Wishing this helps:

Ana

2015-11-29 19:22 GMT+01:00 José Manuel Otero Dapena <
notifications@github.com>:

> Well, talking about relations in Odoo, for each product there are one and
> only one bill of materials.
> 
> [image: 2015-11-29 19_13_36-sin titulo - paint]
> https://cloud.githubusercontent.com/assets/15608043/11459079/72676c40-96ce-11e5-81b9-22437860722e.jpg
> 
> A Bill of Row Materials (BoM) can contain one o more Row Material, but
> only one Routing (production line), we could say that BoM is one table.
> Whose line of production is only and its primary key is the product. The
> BoM contains one foreing key for each row material an only one for routing
> (contains one foreing key for each row material and only one for routing
> (routing, productive line and productive process are the same concept).
> 
> In Odoo, Work Order is a relation between Operation and end product, both
> takes its name. Wok order is performed in a Work Center, and the capacity
> of the Work Center is predefined in itself, unless, otherwise directed by a
> work order.
> 
> Therefore a table is necessary or at least a view to store and then
> display, the actual capacity of the Centre for a work order and the hours
> provided for her in particular.
> 
> For example, if the machinery of a work center consists of a saw, does not
> require the same effort cutting timber that polystyrene foams, and
> therefore has less real capacity for the first thing for the second and
> that means more running time for the same operation, cut.
> 
> Do now, again I ask, what I have to prove?. What test requires the PR
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/manufacture/pull/57#issuecomment-160442214.

## 

CEO Avanzosc, S.L http://www.avanzosc.es : Office phone / Tfono oficina:
(+34) 943 02 69 02
Ana Juaristi Olalde http://www.anajuaristi.com/: Personal phone: 677 93
42 59. User/usuario skype: Avanzosc
www.openerpsite.com

_El contenido de esta comunicación y de toda su documentación anexa es
confidencial y se dirige exclusivamente a su destinatario. El uso no
autorizado de esta información está prohibido por la legislación vigente.
Si usted no es el destinatario le rogamos nos lo indique, no comunique su
contenido a terceros y proceda a su destrucción. Disculpe las molestias que
le haya ocasionado la recepción indebida de este e-mail. Sus datos figuran
en un fichero cuyo titular es Avanzosc, S.L., a quien usted puede dirigirse
para ejercer sus derechos de acceso, rectificación, cancelación y oposición
en Klara Donea 13, 20720, Azkoitia (Gipuzkoa), Tef. 943 02 69 02 -
*_administracion@avanzosc.
soporte@avanzosc.eses*

_Komunikazio honen edukia eta dokumentazio erantsia konfidentziala da eta
hartzaileak bakarrik jaso beharko luke. Indarrean dagoen legeriak debekatu
egiten du bertan eskainitako informazioa baimenik gabe erabiltzea.
Komunikazioa zuri iritsi bazaizu, baina zu ez bazara hartzailea, mesedez,
guri jakinarazi, eta jasotako informazioa ez inori jakinarazi eta suntsitu.
Barkatu okerreko email hau jasotzeak eragindako eragozpenak. Zure datuak
Avanzosc, S.L. enpresaren fitxategietan sartuta daude. Zure datuak atzitzea
eska dezakezu, bai eta, datuak zuzentzea, ezereztea eta tratamenduari aurka
egitea ere. Horretarako, enpresara jo dezakezu, helbide honetan: *_Klara
Donea 13*\* 20720, Azkoitia (Gipuzkoa), telefonoa: 943 02 69 02 - *
_administracion@avanzosc. soporte@avanzosc.eses_ _This message and all
documents attached to it are confidential and intended only for the person
or entity to which it is addressed. Any use of this information by
unauthorised persons is prohibited under current legislation. If you
received this message by error, please advise us, destroy it and refrain
from communicating its contents to third parties. We apologise for any
inconvenience receiving this email improperly may cause to you. Your
personal data are included in a file owned by Avanzosc, S.L. If you want to
exercise your rights of access, correction, erasure and objection you can
contact the Controller at *_Klara Donea 13*\* 20720, Azkoitia (Gipuzkoa), T:
943 02 69 02 – administracion@avanzosc. soporte@avanzosc.eses*

JMODaERP commented:
You're right in the foregoing. 

As we can see is configurable: “Allow several bill of materials per products using properties”. Until now I have not used this feature.

As for the rest, I know that in a customized version several resources (humans and machines) can be used, to the end and both are within productive resources, and is logical that each resource has a different potential capacity. I've tested Odoo Core, Odoo OCA and Customer Odoo and to be several possibilities I've painted the simplest one. 

Your notes have been very useful. I shall remake  the diagram of entities and modify the relations. 

A cordial greeting.

rafaelbn commented:
@anajuaristi @pedrobaeza about @nhomar https://github.com/OCA/manufacture/pull/57#issuecomment-160173847
- Ana do you have any short video? or something similar?
- Pedro, I think that if you review comments, Eva, Ana and me could make las functional review and go ahead with this.

ping @tafaRU

rafaelbn commented:
In mrp_operation_extension's README there is a mistake in configuration:

> Go to Settings > Manufacturing and activate "Manage routings and work orders" for handling the features of this module.
- Activating this option doesn't install mrp_operation_extension

I would recommend for testing: 
1. Start a base database

![2015-12-02_0-37-29](https://cloud.githubusercontent.com/assets/8736623/11517886/022ac204-988d-11e5-82f7-97074a64d456.png)
2. Go To Setting - User and activate technical features in admin user
3. Go To Setting - Modules and install **only**  mrp_operations_extension 
4. Go to Setting - Manufacturing - Planning and enable options
    - Manage routing and work order
    - Allow detailed planning of work order
    - Manage operators in work centers

![2015-12-02_0-50-16](https://cloud.githubusercontent.com/assets/8736623/11518088/ae38aa74-988e-11e5-8f49-f5644d4b7173.png)

ybofr commented:
Hi,
Is the pull request available for testing on OCA? if yes, what should I choose, as I presume, I should not choose OdooMRP anymore.

Thanks for the answer
Eva P.

oihane commented:
Hi @ybofr,
can you see the checks at the end of the PR?

![image](https://cloud.githubusercontent.com/assets/2323616/11525464/9abaaf98-98d3-11e5-8e8f-574ebe5e633c.png)

There is one that says **runbot** if you click on details, you will be redirected to the corresponding OCA runbot

anajuaristi commented:
I would make a sugestion... 
Manage routing and workcenter is an Odoo core check that installs mrp_operations
I think if someone installs directly mrp_operations_extension, that check should be automatically enabled since it has got a dependancy with mrp_operations.

It has nosense installing mrp_operations_extension and don't using routings. 
My2cents

pedrobaeza commented:
@rafaelbn, don't lose the focus: configuration instructions are once you have installed the module, so _mrp_operations_ and _mrp_operations_extension_ are already installed in this point. You only need to set the check for viewing all the stuff.

@anajuaristi, the check doesn't install the module. It simply un-hide some fields and menus.

anajuaristi commented:
Yes, maybe I did not explain very well

I wanted to say that if someone installs operations_extension, that check should be automatically enabled. Otherwise it must to be made manually, since a lot of fields are invisible if it's not enabled. It will improve usability for user. 
Just a sugestion.
.

pedrobaeza commented:
Well, that's up to Odoo that doesn't enable it by default installing _mrp_operations_.

JavierIniesta commented:
Tested in runbot :+1:

JMODaERP commented:
Tested in runbot, :+1:

I am making a little visual procces for explain the proccess to prove.

rafaelbn commented:
@pedrobaeza could you review comment from @elicoidal please?

rafaelbn commented:
Hi @ybofr you could testit here: http://runbot.odoo-community.org/runbot/build/3136482

![02-12-2015 19-07-22](https://cloud.githubusercontent.com/assets/8736623/11539552/f7e8d02e-9927-11e5-9b2a-c7140f392c25.png)

anajuaristi commented:
Tested in
http://runbot.odoo-community.org/runbot/build/3136482

Steps to reproduce:

Create a new product "production"
Creat a new bom for it
Include existing materials on bom
Activate the bom
Go to manufacturing orders from product "production"
Push compute data
Try to confirm
ValidateError

Field(s) `product_uom` failed against a constraint: You try to move a product using a UoM that is not compatible with the UoM of the product moved. Please use an UoM in the same UoM category.

Activate "use different measure units" just to see witch of them is different one.
All materials are set on UOM "unit(s)"

I tried also to set a cost product "service" for machines but I can not find the place where the unit is wrong so I can not confirm the order.

Sorry but having this behavior I have got to say :-1: until we find the wrong product and system gives more clues about how to solve the error.

oihane commented:
@anajuaristi did you test in **-all** database or in **-base** database?

anajuaristi commented:
In the one i provide the link. The same used by @rafaelbn

oihane commented:
@anajuaristi try this link http://3136482-57-9f1f66.runbot1.odoo-community.org/web?db=3136482-57-9f1f66-base because if you have to activate the BoM you are not using the one @rafaelbn has said, there is not installed mrp_bom_version, the main link is the same but you have to choose between two databases.
@ybofr will you test it also?

ybofr commented:
Thanks Rafael

> On 2 Dec 2015, at 18:07, Rafael Blasco notifications@github.com wrote:
> 
> Hi @ybofr https://github.com/ybofr you could testit here: http://runbot.odoo-community.org/runbot/build/3136482 http://runbot.odoo-community.org/runbot/build/3136482
>  https://cloud.githubusercontent.com/assets/8736623/11539552/f7e8d02e-9927-11e5-9b2a-c7140f392c25.png
> —
> Reply to this email directly or view it on GitHub https://github.com/OCA/manufacture/pull/57#issuecomment-161384866.

ybofr commented:
Hi,
Please correct the message for the missing routing. It should be "if you haven't assigned a routing yet..."
<img width="898" alt="bill_of_materials_-_odoo" src="https://cloud.githubusercontent.com/assets/7601880/11553062/09953bae-9983-11e5-9887-9ca065aad3b3.png">

ybofr commented:
Creation of the BOM. Product unit of measure is fixed on unit and is not taken from the product when the product is created on the fly during BOM editing.
In this case, the metal 1 should come in kg

<img width="465" alt="manufacturing_orders_-_odoo" src="https://cloud.githubusercontent.com/assets/7601880/11553136/cb424efe-9983-11e5-8fc6-d761e41b9ef9.png">

ybofr commented:
My comment: test :-1: as the units of measures are not properly taken from the product.

pedrobaeza commented:
@ybofr, @anajuaristi, the features you are trying are not related to these modules. If you're not going to test what this modules does, please don't review it. The things you're finding should be reported as issues of the already merged modules.

JMODaERP commented:
Here is the visual process: 

https://docs.google.com/document/d/15d-VkhxKSHbstM7ONXa41KLh5WmixfKu0su0eCTgMQs/edit?ts=565ee128#

rafaelbn commented:
Resume:
- This PR has 2 x :+1: 
- The 2 x :-1: as commeted by @pedrobaeza are not related as reviewing try to do somethings that this module doesn't do.
- @JMODaERP make a doc to know how to test this module (**thanks a lot for this work**): https://docs.google.com/document/d/15d-VkhxKSHbstM7ONXa41KLh5WmixfKu0su0eCTgMQs/edit?ts=565ee128#
- I will review this module now

If:
- @nhomar is agree with his comments here https://github.com/OCA/manufacture/pull/57#issuecomment-160173847
- @moylop260  is agree with his comments here https://github.com/OCA/manufacture/pull/57#issuecomment-160177905
- @elicoidal cooments reviewed

This module is ready to merge IMHO

rafaelbn commented:
Tested in runbot :+1: Thanks!

anajuaristi commented:
@pedrobaeza 
Just an small bug --> confirm the manufacturing order and before starting one operation change the default machine for another one.
Expected behavior: Recalculate cycles and stimated time for Wo
Resulted behavior: cycles and time are not recalculated, remaning the same

pedrobaeza commented:
OK, let's try to put this in order because there's a lot of noise in this PR:
- @elicoidal, you have made comments in PR https://github.com/OCA/manufacture/pull/60 that belongs here, because as stated in that PR, it includes all this stuff. I have fixed the strings, but I have added a lot more in the README.rst file of _mrp_operations_extension_ module. About `extension` term, I'm afraid in this occasion we cannot use another word nor split the module in parts: it's all attached together as a truly extension for operations. If there are more things needed in operations management, they can be included in the module. Take into account that the boundaries of the module are already very limited.
- @nhomar: I have added tests to increase even more the coverage level. The init hook is safe because it doesn't destroy information. But sorry, no video. I don't like it because they are not "indexable", and I don't have time for make it. I have improved the README file instead.
- @moylop260: I have added images and fixed pylint beta messages.
- @anajuaristi: Try here the error you mention if it still happens.
- @ybofr: The README is improved to see exactly what this module does if you want to test it. There are 2 strategies: remove the other modules of the repo for not interfering, or know what makes each of them for ignoring their behaviours.
- @rafaelbn, @JavierIniesta, @JMODaERP, you can make last test before merging.

elicoidal commented on mrp_operations_extension/README.rst:
```diff
@@ -0,0 +1,135 @@
+============
+
+* *hr_timesheet* is a big dependency for only catching the employee cost when
+  selectiong operators in the workcenter. A glue module for this operation
```
s/selectiong/selecting

pedrobaeza commented:
@elicoidal, fixed

JavierIniesta commented:
Runbot is returning a "504 Gateway Time-out" error.

pedrobaeza commented:
Runbot rebuilding

JavierIniesta commented:
Tested in local environment :+1:

JMODaERP commented:
I can jot creat the BoM in this runbot, therefore i can not do the comntrols. Why?. :-1:

JMODaERP commented:
I can not creat the BoM in this runbot, therefore i can not do the comntrols. Why?.

pedrobaeza commented:
Because runbot already has _mrp_bom_version_ installed. Uninstall that module or activate the bom.

rafaelbn commented:
Tested in runbot :+1:

pedrobaeza commented:
OK, this PR is blocking a lot more, so I'm going to merge with the positive review of 2 reviewers, and with the reasoning behind the thumb down explained. If there's any problem, we can make a PR later.

JMODaERP commented:
Tested and O.K. again. :+1:
